Improve pre-merge checklist (#1526)
kyma-project/istio: .github/pull-request-template.md

**Pre-Merge Checklist**

- [ ] As a PR reviewer, verify code coverage and evaluate if it is acceptable
- [ ] Manifests are updated

> - [ ] For the changes in the PR, consider following:

Reviewer: All the statements in this section seem a little vague, is this on purpose?

> - [ ] For the changes in the PR, consider following:

Reviewer: ```suggestion
- [ ] For the changes in the PR, consider the following topics. If any of this are violated in some way please document in the PR description why and what for.
```

>   - [ ] Old clusters are in the consistent state with the new clusters

Reviewer: ```suggestion
  - [ ] Clusters created before this changed and upgraded behave consistently with newly created clusters
```

  - [ ] Change works on the all hyperscalers
  - [ ] Change does not affect LoadBalancers

>   - [ ] Hyperscaler costs

Reviewer: What tooling should be used to estimate those costs? What should be the target?

>   - [ ] RBAC changes are not too permissive

Reviewer: ```suggestion
  - [ ] RBAC settings are as restrictive as possible
```

>   - [ ] If any new libraries are added, security, license compliance and maintainability

Reviewer 2: ```suggestion
  - [ ] If any new libraries are added, security, license compliance and maintainability
- [ ] Create RN for introduced changes 
  
```

>   - [ ] If any new libraries are added, security, license compliance and maintainability

Reviewer: >  - [ ] If any new libraries are added, security, license compliance and maintainability

How / what tools to use to evaluate?

**Pre-Merge Checklist**

> Consider all the following items. If any of them are violated, or you are not sure about them, please add a comment to the PR.

Reviewer 3: ```suggestion
Consider all the following items. If your contribution violates any of them, or you are not sure about it, add a comment to the PR.
```

- [ ] Verify code coverage and evaluate if it is acceptable.

> - [ ] Create RN for introduced changes.

Reviewer 3: ```suggestion
- [ ] Create release notes for introduced changes.
```

- [ ] If Kubebuilder changes were made, run `make generate-manifests` and commit the changes before merge.

> - [ ] Ensure whether pre-existing managed resources are correctly handled.

Reviewer 3: ```suggestion
- [ ] Ensure that pre-existing managed resources are correctly handled.
```

- [ ] Change works on all hyperscalers supported by BTP Kyma Runtime.
- [ ] There is no upgrade downtime.

> - [ ] For infrastructure changes, check the cost in the hyperscaler.

Reviewer 3: Or maybe: If you made infrastructure changes, check if they increase/affect the hyperscaler's costs.

> - [ ] RBAC settings are as restrictive as possible.

Reviewer 3: ```suggestion
- [ ] Ensure that RBAC settings are as restrictive as possible.
```

- [ ] If any new libraries are added, verify license compliance and maintainability and make a comment in the PR with details. We only allow to include stable releases into the project.

> - [ ] If this change should be cherry-picked to active release branches.

Reviewer 3: ```suggestion
- [ ] Check if this change should be cherry-picked to active release branches.
```

> - [ ] If the change of the configuration does not introduce any additional latency.

Reviewer 3: ```suggestion
- [ ] Check if the change of the configuration does not introduce any additional latency.
```
